// @mralexandernickel/gh-boilerplate/dist/app/services/ScrollShrinkService.js
"use strict";
var ScrollShrinkService = (function () {
    function ScrollShrinkService() {
        this.contentEl = document;
        this.scrollCheckInterval = 100;
        this.showShadowOffset = 20;
        this.didScroll = false;
        this.classShowShadow = 'show-shadow';
        this.headerEl = document.querySelector('header.gh-boilerplate');
        this.init();
    }
    ScrollShrinkService.prototype.init = function () {
        var _this = this;
        if (document.body.scrollTop > this.showShadowOffset) {
            this.headerEl.classList.add(this.classShowShadow);
        }
        this.contentEl.addEventListener('scroll', function () {
            _this.didScroll = true;
        });
        setInterval(function () {
            if (_this.didScroll) {
                _this.didScroll = false;
                if (document.body.scrollTop > _this.showShadowOffset) {
                    _this.headerEl.classList.add(_this.classShowShadow);
                }
                else {
                    _this.headerEl.classList.remove(_this.classShowShadow);
                }
            }
        }, this.scrollCheckInterval);
    };
    return ScrollShrinkService;
}());
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = ScrollShrinkService;
