// @mralexandernickel/gh-boilerplate/dist/app/services/StylesheetsService.js
"use strict";
var CONSTANTS = require('../configs/Constants');
var StylesheetsService = (function () {
    function StylesheetsService() {
        this.head = document.head;
        this.stylesheets = CONSTANTS.STYLESHEETS;
        this.stylesheetsLoaded = 0;
        this.classLoaded = 'css-ready';
        this.get();
    }
    StylesheetsService.prototype.injectStylesheet = function (url) {
        var _this = this;
        var resource = document.createElement('link');
        resource.setAttribute('rel', 'stylesheet');
        resource.setAttribute('href', url);
        resource.addEventListener('load', function () {
            _this.stylesheetsLoaded++;
            if (_this.stylesheetsLoaded === _this.stylesheets.length) {
                document.body.classList.add(_this.classLoaded);
            }
        });
        this.head.appendChild(resource);
    };
    StylesheetsService.prototype.get = function () {
        for (var i = 0; i < this.stylesheets.length; i++) {
            this.injectStylesheet(this.stylesheets[i]);
        }
    };
    return StylesheetsService;
}());
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = StylesheetsService;
